docs/plugins/plugin-writer/first-plugin.rst

```diff
@@ -1,2 +1,109 @@
+
+Check our Example plugin implementation of `the ExampleContent <https://github.com/pulp/pulp_example/blob/master/pulp_example/app/models.py#L87-L114>`_ and its `serializer <https://github.com/pulp/pulp_example/blob/master/pulp_example/app/serializers.py#L7-L13>`_ and `viewset <https://github.com/pulp/pulp_example/blob/master/pulp_example/app/viewsets.py#L13-L17>`_.
+For a general reference for serializers and viewsets, check `DRF documentation <http://www.django-rest-framework.org/api-guide/viewsets/>`_.
+
+
+.. _define-importer:
+
+Define your plugin Importer
+---------------------------
+
+To define a new importer, e.g. ``ExampleImporter``:
+ * :class:`pulpcore.plugin.models.Importer` should be subclassed and extended with additional attributes to the plugin needs,
+ * define ``TYPE`` class attribute which is used for filtering purposes,
+ * ``sync`` method should be defined on a plugin importer model ``ExampleImporter``,
+ * create a serializer for your new importer as a subclass of :class:`pulpcore.plugin.serializers.ImporterSerializer`,
+ * create a viewset for your new importer as a subclass of :class:`pulpcore.plugin.viewsets.ImporterViewSet`.
+
+:class:`~pulpcore.plugin.models.Importer` model should not be used directly anywhere in plugin code. Only plugin-defined Importer classes are expected to be used.
+
+One of the ways to perform synchronization:
+ * Download and analyze repository metadata from a remote source.
+ * Decide what needs to be added to repository or removed from it.
+ * Associate already existing content to a repository by creating an instance of :class:`~pulpcore.plugin.models.RepositoryContent` and saving it.
+ * Remove :class:`~pulpcore.plugin.models.RepositoryContent` objects which were identified for removal.
+ * For every content which should be added to Pulp create but do not save yet:
+
+  * instance of ``ExampleContent`` which will be later associated to a repository.
+  * instance of :class:`~pulpcore.plugin.models.ContentArtifact` to be able to create relations with the artifact models.
+  * instance of :class:`~pulpcore.plugin.models.RemoteArtifact` to store information about artifact from remote source and to make a relation with  :class:`~pulpcore.plugin.models.ContentArtifact` created before.
+
+ * If a remote content should be downloaded right away (aka ``immediate`` download policy), use one of the suggested download solutions (:ref:`concurrent.futures <futures-docs>` or :ref:`asyncio <asyncio-docs>` approach) to start downloading. If content should be downloaded later (aka ``on_demand`` or ``background`` download policy), feel free to skip this step.
+ * Save all artifact and content data in one transaction:
+
+  * in case of downloaded content, create instance of :class:`~pulpcore.plugin.models.Artifact` which refers to a downloaded file on a filesystem and contains calculated checksums for it.
```

Comment:
s/create/create an/

Reply:
Actually should this be:  s/create/create and save an/ ? We want them to save this here right?